lib/Parse/ParseType.cpp

```diff
@@ -1372,29 +1373,29 @@ bool Parser::isOptionalToken(const Token &T) const {
   // A postfix '?' by itself is obviously optional.
   if (T.is(tok::question_postfix))
     return true;
-  
+
   // A postfix or bound infix operator token that begins with '?' can be
-  // optional too. We'll munch off the '?', so long as it is left-bound with
+  // optional too.
+  if (!T.is(tok::oper_postfix) && !T.is(tok::oper_binary_unspaced)) {
+    return false;
+  }
+  // We'll munch off the '?', so long as it is left-bound with
   // the type (i.e., parsed as a postfix or unspaced binary operator).
-  if ((T.is(tok::oper_postfix) || T.is(tok::oper_binary_unspaced)) &&
-      T.getText().startswith("?"))
-    return true;
-  return false;
+  return T.getText().startswith("?"));
 }
 
 bool Parser::isImplicitlyUnwrappedOptionalToken(const Token &T) const {
   // A postfix '!' by itself, or a '!' in SIL mode, is obviously implicitly
   // unwrapped optional.
   if (T.is(tok::exclaim_postfix) || T.is(tok::sil_exclamation))
     return true;
-  // A postfix or bound infix operator token that begins with '!' can be
-  // implicitly unwrapped optional too. We'll munch off the '!', so long as it
-  // is left-bound with the type (i.e., parsed as a postfix or unspaced binary
-  // operator).
-  if ((T.is(tok::oper_postfix) || T.is(tok::oper_binary_unspaced)) &&
-      T.getText().startswith("!"))
-    return true;
-  return false;
+
+  if (!T.is(tok::oper_postfix) && !T.is(tok::oper_binary_unspaced)) {
+    return false;
+  }
+  // We'll munch off the '?', so long as it is left-bound with
+  // the type (i.e., parsed as a postfix or unspaced binary operator).
+  return T.getText().startswith("?"));
```

Comment:
And here

Reply:
And this function had its body copied and pasted. It needs to match the `!`-munging behavior it used to have.